server/src/main/java/com/exacaster/lighter/application/sessions/SessionService.java

```diff
@@ -85,8 +107,12 @@ private Application createSession(SubmitParams params, String sessionId, Applica
     }
 
     protected List<Application> fetchRunningSession() {
-        return applicationStorage
+        var regularSessions = applicationStorage
                 .findApplicationsByStates(ApplicationType.SESSION, ApplicationState.runningStates(), SortOrder.ASC, 0, Integer.MAX_VALUE);
+        var permanentSessions = applicationStorage
```

Comment:
`fetchRunningSession()` is used in scheduled `handleTimeout` method, which is responsible for killing timed-out sessions.
We should never kill permanent sessions.